constants/src/lib.rs

```diff
@@ -186,6 +186,10 @@ pub const INVALID_CHARACTER_VALUE: OdbcState<'static> = OdbcState {
     odbc_2_state: "22005",
     odbc_3_state: "22018",
 };
+pub const INVALID_TRANSACTION_OPERATION_CODE: OdbcState<'static> = OdbcState {
+    odbc_2_state: "HY012",
```

Comment:
Here is the mapping :)
https://learn.microsoft.com/en-us/sql/odbc/reference/develop-app/sqlstate-mappings?view=sql-server-ver17
ODBC 2 is `S1012`

Reply:
nice!